johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mapper.java

```diff
@@ -138,10 +142,15 @@ public void writeObject(final Object object, final Writer stream) {
     private boolean isDeduplicateObjects(Class<?> rootType) {
         Boolean dedup = config.isDeduplicateObjects();
         if (dedup == null) {
-            Mappings.ClassMapping classMapping = mappings.findOrCreateClassMapping(rootType);
-            if (classMapping != null) {
-                dedup = classMapping.isDeduplicateObjects();
+            if (!deduplicateMap.containsKey(rootType)){
+                JohnzonDeduplicateObjects jdo = rootType.getAnnotation(JohnzonDeduplicateObjects.class);
```

Comment:
this should probably move to ClassMapping no?

Reply:
Why?